From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Behrang Fouladi <iyr@firmalyzer.com>
Date: Thu, 20 Apr 2023 16:17:06 +0200
Subject: [PATCH] 5242: fix multiequal op handling in ShowConstantUse.java

process all input varnodes of multiequal pcodeop

remove annotation & clone()
---
 .../Decompiler/ghidra_scripts/ShowConstantUse.java    | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/Ghidra/Features/Decompiler/ghidra_scripts/ShowConstantUse.java b/Ghidra/Features/Decompiler/ghidra_scripts/ShowConstantUse.java
index 3123a67e1..7d50f50b5 100644
--- a/Ghidra/Features/Decompiler/ghidra_scripts/ShowConstantUse.java
+++ b/Ghidra/Features/Decompiler/ghidra_scripts/ShowConstantUse.java
@@ -867,12 +867,11 @@ public class ShowConstantUse extends GhidraScript {
 					doneSet);
 				return;
 			case PcodeOp.MULTIEQUAL:
-				followToParam(constUse, defUseList, highFunction, def.getInput(0), funcList,
-					doneSet);
-				@SuppressWarnings("unchecked")
-				ArrayList<PcodeOp> splitUseList = (ArrayList<PcodeOp>) defUseList.clone();
-				followToParam(constUse, splitUseList, highFunction, def.getInput(1), funcList,
-					doneSet);
+				for (int i = 0; i < def.getNumInputs(); i++) {
+					ArrayList<PcodeOp> splitUseList = new ArrayList<>(defUseList);
+					followToParam(constUse, splitUseList, highFunction, def.getInput(i), funcList,
+						doneSet);
+				}
 				return;
 			case PcodeOp.CAST:
 				followToParam(constUse, defUseList, highFunction, def.getInput(0), funcList,
-- 
2.39.1

